src/edu/csus/ecs/pc2/ui/AutoJudgeStatusFrame.java

```diff
@@ -19,7 +22,7 @@
  */
 
 // $HeadURL$
-public class AutoJudgeStatusFrame extends javax.swing.JFrame implements AutoJudgeNotifyMessages {
+public class AutoJudgeStatusFrame extends javax.swing.JFrame implements AutoJudgeNotifyMessages, IExecuteTimerFrame {
```

Comment:
I am struggling to understand the overall design approach for handling the "execute timer frame".  Specifially, I don't understand the intent behind specifying that an `AutoJudgeStatusFrame implements IExecuteTimerFrame`, as stated in the above line. 

 In OOP parlance (including UML and Java, for instance),  "implements" is a form of _inheritance_; it specifies the "Is-A" relationship.  It doesn't make sense to me to say that `an AutoJudgeStatusFrame is a(n) IExecuteTimerFrame`.  

It appears to me that the relationship ought to be that of the OOP notion of _composition_ ("HAS-A"),  like that which is specified in the code for class `Executable`:  an `Executable`  HAS-A(n) `IExecuteTimerFrame`.  It is declared within the `Executable` class (as a private variable), and it is set by the parameter passed into the `Executable` constructor.  In the same way that it makes sense to say "An `Executable` has an `IExecuteTimerFrame`", it would make sense to say "An `AutoJudgeStatusFrame` HAS-AN `IExecuteTimerFrame`".   

Note that there is a class `ExecuteTimerFrame` which `implements IExecuteTimerFrame`.  This (sort of) makes sense; class `ExecuteTimerFrame` is a concrete implementation of the `IExecuteTimerFrame` interface.  I'm not sure it really makes sense to say the equivalent about class AutoJudgeStatusFrame.  Why does class `Executable` "have-an" `IExecuteTimerFrame` while class `AutoJudgeStatusFrame` "is-an" `IExecuteTimerFrame`?  It seems like there should be _one_ class (`ExecuteTimerFrame`) implementing `IExecuteTimerFrame`, and that one object should be passed to `AutoJudgeStatusFrame` just like it gets passed to `Executable`.  

Note that making such a design change also has the benefit that class `AutoJudgeStatusFrame` would no longer be (effectively) _duplicating_ the methods implemented in `ExecuteTimerFrame`; it would instead call the relevant methods (as needed) in its `IExecuteTimerFrame` member.

I grant that the code as given in the PR may "work" - meaning, it does what is intended from the user point of view.  However, I'm concerned that implementing what (IMHO) appears to be a weird design may in the long run cause us much anguish (when we get around to trying to fix other things, like the AJ execution cycle).  I would much rather see either a redesign at this point, or at least an explanation which counters my arguments above.